src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1413,7 +1413,11 @@ public function resolveEnvPlaceholders($value, $format = null, array &$usedEnvs
             return $result;
         }
 
-        if (!\is_string($value) || 38 > \strlen($value)) {
+        if (
+            !\is_string($value)
+            || 38 > \strlen($value)
+            || !\preg_match('/env[_(]/m', $value)
```

Comment:
Let's use str_contains(env) only and not check the underscore nor the bracket for simplicity?

Reply:
Oh but the check should be case insensitive, so stripos()

Reply:
If only the string "env" is checked, then such values will be checked against containing placeholders
"Removed service "container.env"; reason: abstract.",
"Twig\\Environment",
"kernel.environment",
and many others

which definitely not contains environment placeholders.

In my project difference between my code and stripos approximally 5 seconds

Reply:
ok, then this should be `!preg_match('/env[_(]/i', $value)`
please also keep everything on one line.

Reply:
done

Reply:
@nicolas-grekas is enough or something else?